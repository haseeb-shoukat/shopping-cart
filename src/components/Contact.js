import { useEffect } from "react";
import backgroundImage from "../images/bg.svg";

const Contact = () => {
  useEffect(() => {
    document.body.style.backgroundImage = `url(${backgroundImage})`;
    document.querySelector(".active").classList.remove("active");
    document.querySelector("#contact").classList.add("active");
  }, []);

  return (
    <div className="contact-card">
      <div className="contact-icons">
        <a href="https://github.com/haseeb-shoukat">
          <i className="fa fa-github contact-icon" aria-hidden="true"></i>
        </a>
        <a href="https://www.youtube.com/watch?v=dQw4w9WgXcQ">
          <i className="fa fa-facebook contact-icon" aria-hidden="true"></i>
        </a>
        <a href="https://www.youtube.com/watch?v=dQw4w9WgXcQ">
          <i className="fa fa-instagram contact-icon" aria-hidden="true"></i>
        </a>
        <a href="https://www.youtube.com/watch?v=dQw4w9WgXcQ">
          <i className="fa fa-linkedin contact-icon" aria-hidden="true"></i>
        </a>
        <a href="https://www.youtube.com/watch?v=dQw4w9WgXcQ">
          <i className="fa fa-twitter contact-icon" aria-hidden="true"></i>
        </a>
      </div>
      <div class="contact-address">
        <p>Knock-Offs</p>
        <p>NM 87131, Albuquerque, New York</p>
      </div>
    </div>
  );
};

export default Contact;
