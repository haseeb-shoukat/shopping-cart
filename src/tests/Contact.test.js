import React from "react";
import { render } from "@testing-library/react";
import Contact from "../components/Contact";

describe("Contact component", () => {
  it("Renders successfully", () => {
    const contact = global.document.createElement("div");
    contact.id = "contact";
    global.document.body.appendChild(contact);
    const { container } = render(<Contact />);
    expect(container).toMatchSnapshot();
  });
});
