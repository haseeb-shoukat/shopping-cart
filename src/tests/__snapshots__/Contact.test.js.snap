// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component Renders successfully 1`] = `
<div>
  <div
    class="contact-card"
  >
    <div
      class="contact-icons"
    >
      <a
        href="https://github.com/haseeb-shoukat"
      >
        <i
          aria-hidden="true"
          class="fa fa-github contact-icon"
        />
      </a>
      <a
        href="https://www.youtube.com/watch?v=dQw4w9WgXcQ"
      >
        <i
          aria-hidden="true"
          class="fa fa-facebook contact-icon"
        />
      </a>
      <a
        href="https://www.youtube.com/watch?v=dQw4w9WgXcQ"
      >
        <i
          aria-hidden="true"
          class="fa fa-instagram contact-icon"
        />
      </a>
      <a
        href="https://www.youtube.com/watch?v=dQw4w9WgXcQ"
      >
        <i
          aria-hidden="true"
          class="fa fa-linkedin contact-icon"
        />
      </a>
      <a
        href="https://www.youtube.com/watch?v=dQw4w9WgXcQ"
      >
        <i
          aria-hidden="true"
          class="fa fa-twitter contact-icon"
        />
      </a>
    </div>
    <div
      class="contact-address"
    >
      <p>
        Knock-Offs
      </p>
      <p>
        NM 87131, Albuquerque, New York
      </p>
    </div>
  </div>
</div>
`;
